homeassistant/components/alert/entity.py

```diff
@@ -195,9 +195,14 @@ async def async_turn_on(self, **kwargs: Any) -> None:
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Async Acknowledge alert."""
-        LOGGER.debug("Acknowledged Alert: %s", self._attr_name)
-        self._ack = True
-        self.async_write_ha_state()
+        if self._can_ack:
+            LOGGER.debug("Acknowledged Alert: %s", self._attr_name)
+            self._ack = True
+            self.async_write_ha_state()
+        else:
+            raise ServiceValidationError(
+                "Cannot acknowledge alert with can_acknowledge set to False"
+            )
```

Comment:
Let's rewrite this as a guard

```suggestion
        if not self._can_ack:
            raise ServiceValidationError(
                "Cannot acknowledge alert with can_acknowledge set to False"
            )
        self._ack = True
        self.async_write_ha_state()
```

Reply:
Additionally, since this error shows up in the UI, we need to make it non-technical.

For example: `This alert cannot be acknowledged`.